### PR TITLE
Removed bad public dir and changed all links

### PR DESCRIPTION
**Pull request by bidoubiwa**
There were two public folders. 

One at the root of the documentation, on a the root of the .vuepress folder.
The one at the root of the documentation is not recognized by vuepress as a public folder. Thus any image that was stored inside that directory is accessed using the complete relative path. Which was often something like this:

```
../../public/image
```

I have moved them to the right `public` folder and changed all links.

### CODE REVIEW

File: guides/main_concepts/relevancy.md [hunk 1]
```diff
@@ -46,7 +46,7 @@ Results are sorted by **the similarity of the matched words with the query words
 
 ::: tab Typo
 
-![Image from alias](../../public/image/vogli3.png)
+![Image from alias](/image/vogli3.png)
```
Comment (curquiza) [inline guides/main_concepts/relevancy.md:49]: Should we keep the name `image` since we almost only have images in `.vuepress/public`?

Comment (eskombro) [reply guides/main_concepts/relevancy.md:49]: I think it makes sense. We may have something different than images at some point, and also it makes really clear what's the content of the folder, so why not?

Comment (bidoubiwa) [reply guides/main_concepts/relevancy.md:49]: I will change the name of the folder to something more explanatory

Comment (eskombro) [reply guides/main_concepts/relevancy.md]: I think it makes sense. We may have something different than images, and also it makes really clear what's the content of the folder, so I would be ok with leaving it there

Comment (bidoubiwa) [reply guides/main_concepts/relevancy.md]: So i have changed image to ranking-rules and react-guide and put the appropriate images inside the appropriate folders

Comment (eskombro) [reply guides/main_concepts/relevancy.md]: No that I want to add to your todo list, but what if we do a `references` directoy, a `guides` dir and a `ressources` dir, reflecting the docs structure. (This was inspired by a conversation with @curquiza but I'm not sure thats exactly what she has in mind)

Comment (bidoubiwa) [reply guides/main_concepts/relevancy.md]: This could be done in the long run but I don't see the necessity right now

**Pull request merged**